src/app.rs

```diff
@@ -109,11 +109,17 @@ impl event::EventHandler for App {
                     ActionType::AddItem(item) => {
                         if item == ItemType::Image {
                             // todo: remove unwrap
```

Comment:
Below, in the if-elsif... is some duplicated code. I chose to do it this way, because the solution I came up with to factor-out the duplicated code looked more complicated and introduced a 2nd if-elsif. But, I'm open to amending this.